src/components/booking/BookingForm.tsx

```diff
@@ -0,0 +1,78 @@
+// BookingForm.tsx
+import React, { useState } from 'react';
+import { Box, Stack } from '@mui/material';
+import { Dayjs } from 'dayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import ButtonPrimary from '../../components/button/ButtonPrimary';
+import { BookingFormProps } from '../../types/booking.types';
+
+const BookingForm: React.FC<BookingFormProps> = ({ onSubmit, data }) => {
+  const [selectedDates, setSelectedDates] = useState<[Dayjs | null, Dayjs | null]>([null, null]);
+
+  const { availableDates, accommodationId } = data || {};
+
+  const shouldDisableDate = (date: Dayjs) => {
+    const dateString = date.format('YYYY-MM-DD');
+    return !availableDates?.some((range) => dateString >= range[0] && dateString <= range[1]);
+  };
+
+  const handleDateChange = (index: number) => (newValue: Dayjs | null) => {
+    setSelectedDates((prevSelectedDates) => {
+      const newSelectedDates = [...prevSelectedDates];
+      newSelectedDates[index] = newValue;
+      return newSelectedDates as [Dayjs | null, Dayjs | null];
+    });
+  };
+
+  const handleReserveButtonClick = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (selectedDates[0] && selectedDates[1]) {
+      const reservationData = {
+        startDate: selectedDates[0]?.format('YYYY-MM-DD') || '',
+        endDate: selectedDates[1]?.format('YYYY-MM-DD') || '',
```

Comment:
create file in constants folder time.ts and import format string for it. Check, please if there in project some more such constants